arwen/host/asyncCall.go

```diff
@@ -166,7 +168,9 @@ func (host *vmHost) sendCallbackToCurrentCaller() error {
 		return err
 	}
 
-	metering.UseGas(metering.GasLeft())
+	gasLeft := metering.GasLeft()
+	metering.ForwardGas(gasLeft)
```

Comment:
forward gas could simply use gas inside.

Reply:
Under the current design, `metering.ForwardGas()` only marks some gas as "forwarded", but the gas itself was already consumed on the Wasmer instance.

In this specific case, for `asyncCall()`, gas forwarding happens when the gas is consumed, but this only happens for cross-shard calls. For in-shard calls, gas is used separately by `host.executeSmartContractCall()`, and later is marked as "forwarded" in `host.finishExecuteOnSame/DestContext()". But the gas marked as "forwarded" is not the entire gas - it's the actual used gas, after unspent gas is restored to the caller.